libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -178,12 +212,31 @@ bool HTTPClient::beginInternal(String url, const char* expectedProtocol)
 
 bool HTTPClient::begin(String host, uint16_t port, String uri)
 {
+    DEBUG_HTTPCLIENT("[HTTP-Client][begin] host: %s port: %d uri: %s\n", host.c_str(), port, uri.c_str());
     clear();
+
+    // Remove protocol if any
```

Comment:
This method takes `host` as an argument, i.e. just the DNS name or IP address. It should not expect to see a protocol prefix.